apps/mobile/src/libs/components/input/input.tsx

```diff
@@ -24,30 +32,91 @@ type Properties<T extends FieldValues> = {
 const Input = <T extends FieldValues>({
 	control,
 	errors,
+	isAutoFocused = false,
 	isSecureTextEntry = false,
 	label,
 	name,
 	placeholder,
 }: Properties<T>): JSX.Element => {
+	const [secureTextEntry, setSecureTextEntry] = useState(isSecureTextEntry);
+	const [isFocused, setIsFocused] = useState(false);
+
 	const { field } = useFormController({ control, name });
 
-	const { onBlur, onChange, value } = field;
+	const { onChange, value } = field;
 
 	const error = errors[name]?.message;
 	const hasError = Boolean(error);
+	const FIRST_COLOR_STOP = -0.4;
+	const SECOND_COLOR_STOP = 0.9;
 
+	const borderColors = isFocused
+		? [...GradientColor.BLUE]
+		: [BaseColor.LIGHT_GRAY, BaseColor.LIGHT_GRAY];
+	const conditionalColors = hasError
+		? [BaseColor.RED, BaseColor.RED]
+		: borderColors;
 	return (
 		<View>
-			<Text>{label}</Text>
-			<TextInput
-				onBlur={onBlur}
-				onChangeText={onChange}
-				placeholder={placeholder}
-				secureTextEntry={isSecureTextEntry}
-				style={styles.input}
-				value={value}
-			/>
-			<Text>{hasError && (error as string)}</Text>
+			<Text color={BaseColor.BLACK} size="md" weight="semiBold">
+				{label}
+			</Text>
+			<LinearGradient
+				angle={305}
+				angleCenter={{ x: 0.5, y: 0.5 }}
+				colors={conditionalColors}
+				locations={[FIRST_COLOR_STOP, SECOND_COLOR_STOP]}
+				style={[
+					globalStyles.alignItemsCenter,
+					globalStyles.justifyContentCenter,
+					globalStyles.p2,
+					styles.rounded,
+				]}
+				useAngle
+			>
+				<View
+					style={[
+						globalStyles.flexDirectionRow,
+						styles.input,
+						styles.rounded,
+						globalStyles.alignItemsCenter,
+						globalStyles.pr16,
+					]}
+				>
+					<TextInput
+						autoFocus={isAutoFocused}
+						onBlur={() => {
+							setIsFocused(false);
+						}}
+						onChangeText={onChange}
+						onFocus={() => {
+							setIsFocused(true);
+						}}
+						placeholder={placeholder}
+						placeholderTextColor={BaseColor.LIGHT_GRAY}
+						secureTextEntry={secureTextEntry}
+						style={[
+							globalStyles.justifyContentCenter,
+							globalStyles.alignItemsCenter,
+							globalStyles.pl16,
+							globalStyles.flex1,
+							styles.text,
+						]}
+						value={value}
+					/>
+					{isSecureTextEntry && (
+						<IconButton
+							color={BaseColor.LIGHT_GRAY}
+							name={secureTextEntry ? "visibility-off" : "visibility"}
+							onPress={() => {
+								setSecureTextEntry(!secureTextEntry);
+							}}
+							size={20}
+						/>
```

Comment:
This icon shouldn't depend on isSecureTextEntry. Add the `accessoryRight` property to Input and pass the icon directly.

Reply:
Got stuck with this change. If I understand it clear, I should pass IconButton component as an accessoryRight property and pass props to that component inside an Input element?